flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/DynamicKafkaRecordSerializationSchema.java

```diff
@@ -144,14 +147,36 @@ public void open(
         valueSerialization.open(context);
     }
 
+    private String getTargetTopic(RowData element) {
+        if (topics != null && topics.size() == 1) {
+            // If topics is a singleton list, we only return the provided topic.
+            return topics.get(0);
+        }
+        final String topic = readMetadata(element, KafkaDynamicSink.WritableMetadata.TOPIC);
+        if (topic == null && topics == null) {
```

Comment:
For an empty list, all `topic` metadata will not be allowed. Though I'm not sure how you'd specify that since `""` -> `null`, I guess the user might specify something like `";"`